impl/src/settings.rs

```diff
@@ -395,78 +463,304 @@ fn validate_settings(settings: &mut RazeSettings) -> Result<(), RazeError> {
+  pub workspace_path: Option<String>,
+  #[serde(default)]
+  pub package_aliases_dir: Option<String>,
+  #[serde(default)]
+  pub target: Option<String>,
+  #[serde(default)]
+  pub targets: Option<Vec<String>>,
+  #[serde(default)]
+  pub binary_deps: HashMap<String, cargo_toml::Dependency>,
+  #[serde(default)]
+  pub crates: HashMap<String, CrateSettingsPerVersion>,
+  #[serde(default)]
+  pub gen_workspace_prefix: Option<String>,
+  #[serde(default)]
+  pub genmode: Option<GenMode>,
+  #[serde(default)]
+  pub output_buildfile_suffix: Option<String>,
+  #[serde(default)]
+  pub default_gen_buildrs: Option<bool>,
+  #[serde(default)]
+  pub registry: Option<String>,
+  #[serde(default)]
+  pub index_url: Option<String>,
+  #[serde(default)]
+  pub incompatible_relative_workspace_path: Option<bool>,
+  #[serde(default)]
+  pub rust_rules_workspace_name: Option<String>,
+  #[serde(default)]
+  pub vendor_dir: Option<String>,
+}
+
+impl RawRazeSettings {
+  /** Checks whether or not the settings have non-package specific settings specified */
+  fn contains_primary_options(&self) -> bool {
+    self.workspace_path.is_some()
+      || self.package_aliases_dir.is_some()
+      || self.target.is_some()
+      || self.targets.is_some()
+      || self.gen_workspace_prefix.is_some()
+      || self.genmode.is_some()
+      || self.output_buildfile_suffix.is_some()
+      || self.default_gen_buildrs.is_some()
+      || self.registry.is_some()
+      || self.index_url.is_some()
+      || self.incompatible_relative_workspace_path.is_some()
+      || self.rust_rules_workspace_name.is_some()
+      || self.vendor_dir.is_some()
+  }
+}
+
+/** Grows a list with duplicate keys between two maps */
+fn extend_duplicates<K: Hash + Eq + Clone, V>(
+  extended_list: &mut Vec<K>,
+  main_map: &HashMap<K, V>,
+  input_map: &HashMap<K, V>,
+) {
+  extended_list.extend(
+    input_map
+      .iter()
+      .filter_map(|(key, _value)| {
+        // Log the key if it exists in both the main and input maps
+        if main_map.contains_key(key) {
+          Some(key)
+        } else {
+          None
+        }
+      })
+      .cloned()
+      .collect::<Vec<K>>(),
+  );
+}
+
+/** Parse [`RazeSettings`](crate::settings::RazeSettings) from workspace metadata */
+fn parse_raze_settings_workspace(
+  metadata_value: &serde_json::value::Value,
+  metadata: &Metadata,
+) -> Result<RazeSettings> {
+  let mut settings = RazeSettings::deserialize(metadata_value)?;
+
+  let workspace_packages: Vec<&Package> = metadata
+    .packages
+    .iter()
+    .filter(|pkg| metadata.workspace_members.contains(&pkg.id))
+    .collect();
+
+  let mut duplicate_binary_deps = Vec::new();
+  let mut duplicate_crate_settings = Vec::new();
+
+  for package in workspace_packages.iter() {
+    if let Some(pkg_value) = package.metadata.get("raze") {
+      let pkg_settings = RawRazeSettings::deserialize(pkg_value)?;
+      if pkg_settings.contains_primary_options() {
+        return Err(anyhow!(
```

Comment:
(hopefully this comment doesn't move) 
Here is where error messages for duplicate crate settings (2 package metadata sets contain a definition for the same crate setting) are generated. My though is if you use the same dependency in more than one package, it's a workspace dependency and should be updated to reflect that in raze settings.